extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java

```diff
@@ -68,16 +69,22 @@ ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
     }
 
     @BuildStep
-    NativeImageConfigBuildItem registerAuthClass(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
-        // We must register the auth provider used by the Vert.x mail clients
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true,
-                "io.vertx.ext.mail.impl.sasl.AuthDigestMD5",
-                "io.vertx.ext.mail.impl.sasl.AuthCramSHA256",
-                "io.vertx.ext.mail.impl.sasl.AuthCramSHA1",
-                "io.vertx.ext.mail.impl.sasl.AuthCramMD5",
-                "io.vertx.ext.mail.impl.sasl.AuthDigestMD5",
-                "io.vertx.ext.mail.impl.sasl.AuthPlain",
-                "io.vertx.ext.mail.impl.sasl.AuthLogin"));
+    NativeImageConfigBuildItem registerAuthClass(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+
+        if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
+                x -> x.getGroupId().equalsIgnoreCase("io.vertx")
+                        && x.getArtifactId().equalsIgnoreCase("vertx-mail-client"))) {
+            // We must register the auth provider used by the Vert.x mail clients
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true,
+                    "io.vertx.ext.mail.impl.sasl.AuthDigestMD5",
+                    "io.vertx.ext.mail.impl.sasl.AuthCramSHA256",
+                    "io.vertx.ext.mail.impl.sasl.AuthCramSHA1",
+                    "io.vertx.ext.mail.impl.sasl.AuthCramMD5",
+                    "io.vertx.ext.mail.impl.sasl.AuthDigestMD5",
+                    "io.vertx.ext.mail.impl.sasl.AuthPlain",
+                    "io.vertx.ext.mail.impl.sasl.AuthLogin"));
```

Comment:
The logic will be more complex than that since `AuthCram` seems to replace `AuthCramSHA256`, `AuthCramSHA1`, and `AuthCramMD5` and `AuthDigest` seems to replace `AuthDigestMD5` and `AuthDigestMD5 ` starting from 3.8.0. Even if the versions 3.8.0- are no more supported, the name of the classes have changed as you can see [here](https://github.com/vert-x3/vertx-mail-client/tree/5817da6d218ace197706383744cd73f32d9c8f12/src/main/java/io/vertx/ext/mail/impl/sasl)

Reply:
Interesting, so do you think it's worth adding a version check or shall we make these weak (and add the new class names as well)?

Reply:
Knowing that comparing versions is something that [has already been done ](https://github.com/quarkusio/quarkus/blob/main/extensions/schema-registry/confluent/avro/deployment/src/main/java/io/quarkus/confluent/registry/avro/ConfluentRegistryAvroProcessor.java#L51-L59), I tend to say it's worth adding but it is only my humble opinion

Reply:
It looks like the version used is 4.3.6 and we don't have to worry about other versions as the dependency is pulled due to the following dependency tree:

```
[INFO] +- io.quarkus:quarkus-mailer:jar:999-SNAPSHOT:compile
[INFO] |  +- io.quarkus:quarkus-vertx:jar:999-SNAPSHOT:compile
[INFO] |  |  ...
[INFO] |  \- io.smallrye.reactive:smallrye-mutiny-vertx-mail-client:jar:2.29.0:compile
[INFO] |     \- io.vertx:vertx-mail-client:jar:4.3.6:compile
```

the `io.smallrye.reactive:smallrye-mutiny-vertx-mail-client` version is fixed in [the quarkus bom file](https://github.com/quarkusio/quarkus/blob/65c3f7df50da58a1840629ada95f990666e46278/bom/application/pom.xml#L57) and `io.smallrye.reactive:smallrye-mutiny-vertx-mail-client` defines the `io.vertx:vertx-mail-client` version in https://github.com/smallrye/smallrye-mutiny-vertx-bindings/blob/b70234921a911286872df03e1504efad7b2d2319/pom.xml#L49

Reply:
@cescoffier WDYT? Do we still to register these classes for reflection? It looks like they have not been registered for quiet a while...

Shall we register:
```
            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true,
                    "io.vertx.ext.mail.impl.sasl.AuthCram",
                    "io.vertx.ext.mail.impl.sasl.AuthDigest",
                    "io.vertx.ext.mail.impl.sasl.AuthLogin",
                    "io.vertx.ext.mail.impl.sasl.AuthPlain"));
```
for vertx-mail-client 4.3.6 onwards?